modules/mobile/app/models/mobile/v0/adapters/va_facilities.rb

```diff
@@ -5,23 +5,24 @@ module V0
     module Adapters
       class VAFacilities
         def map_appointments_to_facilities(appointments, facilities)
-          facilities_by_id = facilities.index_by(&:id)
+          facilities_by_id = facilities&.index_by(&:id)
 
           appointments.map do |appointment|
-            facility = facilities_by_id["vha_#{appointment.id_for_address}"]
-            if facility.nil?
+            facility = facilities_by_id&.dig("vha_#{appointment.id_for_address}")
+            if facility.nil? &&
```

Comment:
So this condition seems to be the heart of the ticket. In the case of a video connect appointment without a facility, we still want to have a location name. And location requires an address, which is why we're changing the facilities helper. That's leaving us in an odd place where we have a bunch of conditional logic (see all the safety operators here and in the facilities helper). It might be better to be more explicit in our conditions. 
 ```
if facility
  # create appointment as normal
elsif appointment.appointment_type == VAAppointments::APPOINTMENT_TYPES[:va_video_connect_home]
  # create appointment with name 'No location provided' and a bunch of nils
else
  # raise error
end
 ```

Reply:
yes, I agree. I'll update this.